model_tools/activations/pca.py

```
@@ -75,7 +75,10 @@ def init_and_progress(layer, activations):
     @classmethod
     def hook(cls, activations_extractor, n_components):
         hook = LayerPCA(activations_extractor=activations_extractor, n_components=n_components)
-        assert not cls.is_hooked(activations_extractor), "PCA already hooked"
+        hooks = activations_extractor._extractor._batch_activations_hooks
+        for key in list(hooks):
+            if isinstance(hooks[key], cls):
+                del hooks[key]
```

Isn't this case indicative of a bug? Someone has already hooked PCA and is hooking it again? I would raise an Error here.

---

Also reverted